### DISCUSSION TITLE
ComputeFiniteStrainElasticStress and incremental strain

### DISCUSSION THREAD

**Discussion by MusannaGalib** [idaholab/moose#25429] [Q&A Modules: Solid mechanics]
1. Does MOOSE use hypoelastic (incremental strain) formulation in their **ComputeFiniteStrain** kernel?

2.  Regarding implementing the **ComputeFiniteStrainElasticStress** kernel, which material model does it use (constitutive model)? 

3.  How does the frame of reference (Lagrangian/Eulerian) play a role in the incremental strain approach?

4.  In the tensor mechanics module, is only the static equilibrium equation solved or the full dynamic equation solved?

**Comment by jiangwen84** [comment]
For the incremental formulation, MOOSE implementation follows this paper " MM Rashid. Incremental kinematics for finite element applications. International Journal for Numerical Methods in Engineering, 36(23):3937–3956, 1993. "

Most of your questions can be answered by that paper. Here are some short answers:

> * Does MOOSE use hypoelastic (incremental strain) formulation in their **ComputeFiniteStrain** kernel?
Yes, incremental for hypoelastic. 

> * Regarding implementing the **ComputeFiniteStrainElasticStress** kernel, which material model does it use (constitutive model)?
HypoElastic. 

> * How does the frame of reference (Lagrangian/Eulerian) play a role in the incremental strain approach?
Eulerian.

> * In the tensor mechanics module, is only the static equilibrium equation solved or the full dynamic equation solved?
Both, depending on whether the mass and damping terms are included or not. 

There is a newer implementation of total formulation. See [
](https://mooseframework.inl.gov/source/kernels/lagrangian/TotalLagrangianStressDivergence.html)

**Reply by MusannaGalib** [reply]
Thanks for the reply.

Can you attach the link/documentation here for the implementation of the total formulation?

**Comment by ngrilli** [comment]
@MusannaGalib  examples of input files for dynamics problems are here:

https://mooseframework.inl.gov/modules/tensor_mechanics/Dynamics.html

Basically using DynamicTensorMechanics instead of TensorMechanics
and adding the inertia kernels, you can easily switch from a static
to a dynamic simulation.
